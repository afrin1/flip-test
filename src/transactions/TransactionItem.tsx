import React from 'react'
import { View, Text, StyleSheet } from 'react-native'
import { SuccessButton, PendingButton } from '../components/Button'
import { colors } from '../theme/colors'
import testID from '../utils/testID'
import { Status, Transaction } from './types'

interface TransactionItemProps {
  item: Transaction
  navigation: any
}

const TransactionItem = (props: TransactionItemProps) => {
  const { beneficiaryName,
    senderBank,
    beneficiaryBank,
    amount,
    createdAt,
    status } = props.item
  const buttonText: string = status === Status.SUCCESS ? 'Berhasil' : 'Pengecekan'

  const onButtonPressed = () => {
    props.navigation.navigate('TransactionDetail', {...props.item} )
  }
  
  return (
    <View
      {...testID('transaction-item')}
      style={styles.container}>
      <View {...testID('sidebar')} style={status === Status.SUCCESS
        ? [styles.sideBar, styles.sideBarSuccess]
        : [styles.sideBar, styles.sideBarPending]}></View>
      <View style={styles.item}>
        <View style={styles.content}>
          <Text style={styles.bankName}>{`${senderBank} -> ${beneficiaryBank}`}</Text>
          <Text style={styles.otherInfo}>{beneficiaryName}</Text>
          <Text style={styles.otherInfo}>{`Rp.${amount} . ${createdAt}`}</Text>
        </View>
        {status === Status.SUCCESS
          ? <SuccessButton
            {...testID('success-button')}
            title={buttonText}
            onPress={onButtonPressed} />
          : <PendingButton
            {...testID('pending-button')}
            title={buttonText}
            onPress={onButtonPressed} />}
      </View>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    marginTop: 10,
    flexDirection: 'row'
  },
  sideBar: {
    flex: 0.02,
    borderTopLeftRadius: 10,
    borderBottomLeftRadius: 10,
  },
  sideBarSuccess: {
    backgroundColor: colors.SUCCESS
  },
  sideBarPending: {
    backgroundColor: colors.PENDING
  },
  item: {
    flex: 1,
    padding: 10,
    backgroundColor: 'white',
    flexDirection: 'row',
    borderTopRightRadius: 5,
    borderBottomRightRadius: 5,

  },
  bankName: {
    fontWeight: 'bold',
    fontSize: 20,
  },
  otherInfo: {
    fontSize: 16,
    fontWeight: 'bold'
  },
  content: {
    flex: 1
  }
})

export default TransactionItem