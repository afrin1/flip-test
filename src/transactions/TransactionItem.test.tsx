import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import TransactionItem from './TransactionItem';

it('renders correctly', () => {
  const item = {
    id: "FT2366",
    amount: 2629107,
    uniqueCode: 726,
    status: "SUCCESS",
    senderBank: "bni",
    accountNumber: "8379706095",
    beneficiaryName: "Sammy-Jo Mccall",
    beneficiaryBank: "bca",
    remark: "sample remark",
    createdAt: "2021-09-02 20:49:49",
    completedAt: "2021-09-02 20:49:49",
  }
  renderer.create(<TransactionItem item={item} />);
});

//green color box
//orange box
//success button - testID
//pending button - testId
