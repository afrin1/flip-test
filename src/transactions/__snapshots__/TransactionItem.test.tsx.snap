// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction Item should match snapshot 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginTop": 10,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "borderBottomLeftRadius": 10,
          "borderTopLeftRadius": 10,
          "flex": 0.02,
        },
        Object {
          "backgroundColor": "#00AF33",
        },
      ]
    }
    testID="sidebar"
  />
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderBottomRightRadius": 5,
        "borderTopRightRadius": 5,
        "flex": 1,
        "flexDirection": "row",
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        bni -&gt; bca
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Sammy-Jo Mccall
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Rp.2629107 . 2021-09-02 20:49:49
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "justifyContent": "center",
          "opacity": 1,
        }
      }
      testID="success-button"
    >
      <Text
        style={
          Object {
            "backgroundColor": "#00AF33",
            "borderRadius": 10,
            "color": "#FFFFFF",
            "fontSize": 14,
            "fontWeight": "bold",
            "margin": 5,
            "padding": 10,
            "textAlignVertical": "center",
          }
        }
      >
        Berhasil
      </Text>
    </View>
  </View>
</View>
`;
